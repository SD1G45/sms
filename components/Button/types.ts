export interface ButtonProps extends React.HTMLAttributes<HTMLButtonElement> {
  invert?: boolean;
  disabled?: boolean;
  className?: string;
  loading?: boolean;
}
