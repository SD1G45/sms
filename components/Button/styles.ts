import React from "react";
import styled from "styled-components";
import { ButtonProps } from "./types";

export const StyledButton = styled.button<ButtonProps>`
  width: 100%;
  padding: 12px 14px;
  background-color: ${(props) => (props.invert ? "white" : "#4881F0")};
  border-radius: 18px;
  border: 2px solid ${(props) => (props.invert ? "#4881F0" : "white")};
  color: ${(props) => (props.invert ? "#4881F0" : "white")};
  cursor: ${(props) => (props.disabled ? "not-allowed" : "pointer")};
  font-size: 1rem;
  opacity: ${(props) => (props.disabled ? "40%" : "100%")};
  overflow: hidden;
  white-space: nowrap;

  &:hover {
    opacity: ${(props) => (props.disabled ? "40%" : "70%")};
  }
`;

export const LoadingContainer = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: center;
  gap: 10px;
`;
