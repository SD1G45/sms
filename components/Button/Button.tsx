import React from "react";
import { StyledButton } from "./styles";
import { ButtonProps } from "./types";

const Button: React.FC<ButtonProps> = ({
  invert,
  className,
  children,
  ...other
}) => {
  return (
    <StyledButton invert={invert} className={className} {...other}>
      {children}
    </StyledButton>
  );
};

export default Button;
