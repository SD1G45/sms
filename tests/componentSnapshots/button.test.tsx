import React from 'react';
import renderer from "react-test-renderer";
import "jest-styled-components";
import Button from "../../components/Button";

test('Button renders correctly', () => {
  const tree = renderer.create(<Button/>).toJSON();
  expect(tree).toMatchSnapshot();
});