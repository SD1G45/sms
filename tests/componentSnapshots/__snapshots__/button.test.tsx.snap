// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders as disabled 1`] = `
<button
  className=""
  disabled={true}
/>
`;

exports[`Button renders as loading 1`] = `
<button
  className=""
  loading={true}
>
  <div
    className=""
  >
    <div
      className=""
      color="#fff"
      size={20}
    >
      <div
        className=""
        color="#fff"
        size={20}
      />
    </div>
    <div>
      Loading
    </div>
  </div>
</button>
`;

exports[`Button renders correctly 1`] = `
<button
  className=""
/>
`;
